std/js/_std/Type.hx

```diff
@@ -203,30 +203,55 @@ enum ValueType {
 		if( a == b )
 			return true;
 		try {
+			#if !js_enums_as_objects
 			if( a[0] != b[0] )
 				return false;
 			for( i in 2...a.length )
 				if( !enumEq(a[i],b[i]) )
 					return false;
+			#else
+			if (a._hx_index != b._hx_index)
+				return false;
+			for (f in Reflect.fields(a)){
+				if ( !enumEq(a[f],b[f]) ){
+					return false;
+				}
+			}
+			#end
 			var e = a.__enum__;
 			if( e != b.__enum__ || e == null )
 				return false;
+
 		} catch( e : Dynamic ) {
 			return false;
 		}
 		return true;
 	}
 
 	public inline static function enumConstructor( e : EnumValue ) : String {
+		#if !js_enums_as_objects
 		return untyped e[0];
+		#else
+		return untyped e.__enum__.__constructs__[e._hx_index];
+		#end
 	}
 
 	public inline static function enumParameters( e : EnumValue ) : Array<Dynamic> {
+		#if !js_enums_as_objects
 		return untyped e.slice(2);
+		#else
+		var n = enumConstructor(e);
+		return untyped e.__enum__[n].__params__ ?
```

Comment:
The js Type implementation quite heavily relies on native semantics throughout. The Impression I got is that if Haxe syntax is compatible, it's being used. If that's likely to cause issues down the line, e.g. for the analyzer, then we should use __js__, otherwise I'd prefer not to use it simply because readability suffers a bit due to the __js__ interpolation syntax.